docs/topics/strings.md

```diff
@@ -132,4 +141,36 @@ use the following syntax:
 val price = """
 ${'$'}_9.99
 """
-```
\ No newline at end of file
+```
+
+## String formatting
+To deal with complex formatting requirements, use the [`String.format()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/format.html) 
+function. It uses a format string that defines how the given arguments look like. The format string contains placeholders (`%`) 
+and format specifiers. In the output, the arguments replace the placeholders and show the format that the specifiers define:
+
+```kotlin
+fun main() {
+    //sampleStart
+    // format to add zeroes and make a length of seven
+    val integerNumber = String.format("%07d", 31416)
+    println(integerNumber)
+    // 0031416
+
+    // format with four decimals and sign
+    val floatNumber = String.format("%+.4f", 3.141592)
+    println(floatNumber)
+    // +3.1416
+
+    // format with uppercase for two placeholders
+    val string = String.format("%S %S", "hello", "world")
+    println(string)
+    // HELLO WORLD
+    //sampleEnd
+}
+
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
+
+This function provides similar functionality to string templates. In comparison, the `String.format()` function is 
+more versatile because you have many options for formatting, and it happens during runtime. In contrast, this function is 
```

Comment:
Done. @ilya-g  Can you take a look at the accuracy here? Any other ideas about benefits of formatting strings during runtime are welcome!

Reply:
Technically, string interpolation also happens in run time, e.g. when the template parts are not constants.
What is different is that with `String.format`, the format pattern is not constant and can be taken from a variable.
This can be useful e.g. in localization use cases when you take a localized pattern depending on the user locale. However it is not that flexible and even more error prone as it requires to preserve the placeholder order, which may be challenging if the localized text requires for example a different word order in sentences.

Reply:
Thanks for the explanation, I fixed it.